### ISSUE TITLE
Addon setup - `ponzu get github.com etc`

### ISSUE THREAD

**Issue by olliephillips** [ponzu-cms/ponzu#85]
Just thinking out loud as I to get grips with Ponzu features - would  a `go get` style install of addons be useful - something like this?

```
ponzu get github.com/etc
```

Many code editors offer auto import and code completion. Addons need calling from a content struct to work with, but auto import and code completion will not work, since the addon is not in the go path, if it has only been copied to the `/addons` folder.

This is not a big deal, but it struck me that ``ponzu get`` could wrap ```go get``` adding the addon to the gopath and then additionally copying it to the `/addons` directory. 

With a flag it could be also auto installed in Ponzu project, by writing the import to a file with a prefixed underscore, so you see the admin/addons interface first, before adding any calling code from a content struct file - which is more intuitive. I maybe don't understand the practicalities/side effects of this.

Just wanted to log my ideas as I go, if there is a better way than in an issue please say.

**Comment by nilslice** [comment]
@olliephillips - I like the idea, and agree it would probably be very helpful for quickly getting addons and integrating them into the project.

Hows this for a modified version of your proposal:
```bash
# this would run: go get -u github.com/bosssauce/fbscheduler
# and copy that package into ./addons
$ ponzu add github.com/bosssauce/fbscheduler
```

This way the code completion will work straight away and the addon becomes vendored upon the next build. 

I'm concerned that automatically adding _ imports to some file is a fix for behavior that I'm not sure we want to encourage.. will have to get more feedback on what the downside of this could be. My worry is that without explicitly importing the addon package somewhere, we open up ourselves to having to somehow determine if the _ import should be made (and is safe to do so), or add more API & documentation so the developer knows how to trigger this auto import.. 

I guess its a question of how much "magic" is going on which could end up causing more hurt than good. There is already one instance of this in `cmd/ponzu` here: https://github.com/ponzu-cms/ponzu/blob/master/cmd/ponzu/main.go#L19 but it's truly necessary so that the whole system knows about content types - which are fundamental. Addons could have unexpected side-effects, like you mention.

What if we added a package to the top-level of Ponzu, similar to the following, and move the content package import there replacing the _ "content" import in cmd/ponzu with _ "startup" import:

`startup/run.go`
```go
// Package startup is meant to serve as a place to import packages to trigger 
// their initialization, start delayed workers, etc. `startup` is imported by the
// `cmd/ponzu` main package, and is run when the server starts up
package startup 

import (
    _ "github.com/ponzu-cms/ponzu/content"
    _ "github.com/bosssauce/fbscheduler"
)
```

This offers a place to put addon initialization in the event that an addon isn't tied directly to any one content type, but keeps the auto-initialization explicit (and optional), which I have a feeling will avoid potential issues in the future. Upon building, this startup package would be copied into the vendor directory just like content...

**Comment by olliephillips** [comment]
Yes, `add` is better than `get`. Much better.  I'll have a stab at that feature if you can use - and not time critical?

Re auto-magically importing addons.  Some of what you suggest I will have to digest in the morning - but generally, I think the Addon UX feels a bit odd having to trigger the import in code first, just to see the Addon admin suite in the dashboard.

**Comment by nilslice** [comment]
If you'd like to work on the feature, please go for it! And feel free to loop me in whenever needed. I'm typically in the #ponzu channel on this Slack if you'd like to join and discuss:
https://invite.slack.golangbridge.org

Take note that the CLI code changes should be made to the `ponzu-cms/ponzu` package itself, not a project that you create with `ponzu new` - and there are some contribution docs in the README. You'll need to run `$ go install ./...` from the `ponzu-cms/ponzu` package or if you fork it, from your own fork (then using the `-fork` flag as shown in the README) - sorry if this is all obvious, just trying to leave notes wherever possible so others might find them if they are having trouble contributing.

I would also suggest familiarizing yourself with the `copyAll` func here: https://github.com/ponzu-cms/ponzu/blob/master/cmd/ponzu/options.go#L334, as well as the various places it is called.. I think you'll probably want to use that when copying a `go get` package into `./addons` - the build process should already account for anything added into `./addons`, but it will be something you'll want to verify manually. 

re: addon auto _ imports.. I agree that it feels odd, and maybe we simply document the Addon API in such a way that strongly suggests addon developers only run the `addon.Register` func in the addon package's `init` func - and nothing else (unless they know exactly how it will impact the system). I do plan on creating a generator for addons, much like content, so `$ ponzu gen addon ....` is a way to standardize this a bit. 

If we can do that elegantly and practically, then I would feel much better about doing the auto _ import. Also, we will need to clearly document that the auto _ import only happens when using `$ ponzu add ...` so there is no expectation of it happening when someone simply copy/pastes a package into `./addons`

**Comment by olliephillips** [comment]
Thanks, yes I'll look at this. My pleasure.

**Comment by nilslice** [comment]
Merged #87. Thank you!
